### PR TITLE
✨ Add yaml CI validation for yml\yaml files

### LINKED ISSUE #218: Add yaml CI validation for metadata.yml (and probably all yaml in the repo)
This would allow some basic protection against invalid YAML from landing in the project.

/kind feature

### PR DESCRIPTION
**Pull request by omerap12**
**What this PR does / why we need it**:
This gives us basic protection against invalid YAML from landing in the project. we can configure which types of checks we want to include and which checks should be set as warning.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #218

### CODE REVIEW

File: .github/workflows/yamlvalidator.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    types: [opened, edited, synchronize, reopened]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Lint YAML files
+        run: yamllint -c .github/yamllint_config.yaml .
```
Comment (Jont828) [inline .github/workflows/yamlvalidator.yml:14]: Can we get a newline at the EOF here?

**Review by Jont828 — COMMENTED** [review]
Minor nit but otherwise LGTM

File: .github/workflows/yamlvalidator.yml (hunk 1 above)
Comment (omerap12) [reply .github/workflows/yamlvalidator.yml:14]: Sure

**Comment by Jont828** [comment]
Can you squash commits as well? Should be good to merge we do that and tests are done.

**Comment by omerap12** [comment]
> Can you squash commits as well? Should be good to merge we do that and tests are done.

sure

**Pull request merged**